### PR TITLE
Propagate CLI subprocess exit code

### PR DESCRIPTION
**Pull request by mxr**
#### Problem
The console script currently discards the return code from the Java subprocess. That means callers can see a successful Python CLI exit even when Java fails.

#### Solution
Exit from the console entry point with the subprocess return code. Add a regression test that verifies the CLI propagates a non-zero result.

---
## Summary by cubic
Propagate the Java subprocess exit code from the `openapi-generator-cli` entrypoint so callers and CI fail correctly. `cli` now accepts `argv` and defaults to `sys.argv` to preserve behavior and simplify testing.

- **Bug Fixes**
  - Exit with the Java return code via `sys.exit(run(args).returncode)` and parse args as `(argv or sys.argv)[1:]`.
  - Add tests for `run([...])` and `cli([...])` to assert non-zero exit codes, error message, and no stdout on invalid args.
  - Document the `cli(argv)` argument and default behavior in the docstring.

<sup>Written for commit 44495ecc0169c3dc6b0d3230efb8510d125d1551. Summary will update on new commits. <a href="https://cubic.dev/pr/OpenAPITools/openapi-generator-pip/pull/55?utm_source=github">Review in cubic</a></sup>

### CODE REVIEW

File: tests/test_cli.py
Comment (eggplants) [inline tests/test_cli.py:1]: Instead of mocking, please test that running `openapi-generator-cli this-is-invalid-subcommand` returns an exit code of 1.

**Review by eggplants — CHANGES_REQUESTED** [review]

File: tests/test_cli.py
Comment (mxr) [reply tests/test_cli.py:1]: sure

Comment (mxr) [reply tests/test_cli.py:1]: A bit tricky since `cli()` takes no args. I could mock out `sys.argv` or I could change `cli` to take optional args. What do you prefer? I did the first option for now

File: openapi_generator_cli/__init__.py [hunk 1]
```diff
@@ -56,13 +56,11 @@ def run(args: list[str] | None = None) -> subprocess.CompletedProcess[bytes]:
...
 
 
-def cli() -> None:
+def cli(argv: list[str] | None = None) -> None:
     """Run the OpenAPI Generator CLI with the arguments provided on the command line."""
-    args = []
-    if len(sys.argv) > 1:
-        args = sys.argv[1:]
-    run(args)
+    args = argv[1:] if argv is not None and len(argv) > 1 else []
```
[the remark this answers is not in the document]
Comment (mxr) [reply openapi_generator_cli/__init__.py]: fixed

File: tests/test_cli.py [hunk 2]
```diff
@@ -2,12 +2,10 @@
```
[the remark this answers is not in the document]
Comment (mxr) [reply tests/test_cli.py:2]: don't think this is valid feedback as this is the only `cli()` test

File: openapi_generator_cli/__init__.py
Comment (eggplants) [inline openapi_generator_cli/__init__.py:1]: To be simple:
```python
def cli(argv: list[str] | None = None) -> None:
    """Run the OpenAPI Generator CLI with the arguments provided on the command line."""
    args = (argv or sys.argv)[1:]
    sys.exit(run(args).returncode)


if __name__ == "__main__":
    cli()
```

**Review by eggplants — CHANGES_REQUESTED** [review]

File: openapi_generator_cli/__init__.py
Comment (mxr) [reply openapi_generator_cli/__init__.py:1]: sure

**Review by eggplants — APPROVED** [review]
LGTM. Thank you @mxr!
@wing328 Just in case, please review this.

File: openapi_generator_cli/__init__.py [hunk 3]
```diff
@@ -56,12 +56,10 @@ def run(args: list[str] | None = None) -> subprocess.CompletedProcess[bytes]:
     return subprocess.run(arguments, check=False)  # noqa: S603
 
 
-def cli() -> None:
+def cli(argv: list[str] | None = None) -> None:
     """Run the OpenAPI Generator CLI with the arguments provided on the command line."""
```
Comment (eggplants) [inline openapi_generator_cli/__init__.py]: Add `Args` section to this docstring.

**Review by eggplants — CHANGES_REQUESTED** [review]

**Review by eggplants — APPROVED** [review]
